BEST_PRACTICES.md

```diff
@@ -80,7 +80,45 @@ Keep in mind that jobs can be interrupted at any point. States should never be i
 state. Interrupted jobs that save state too early will have data missing. Interrupted jobs that
 save state too late will cause an increase in duplicate rows being replicated.
 
-The tap's config file must ALWAYS have a `start_date` field indicating the default state.
+The Tap's config file can include a `start_date` field indicating the default state.
+
+### Stream bookmark format
+
+State records serve to indicate a Tap's progress through a data source, but they can also provide
+more granular information about progress through individual streams.
+
+If Tap streams can each have an individual state, the state output by the Tap should conform to the
+following format: the state object contains a top-level property `"bookmarks"` that maps to an
+object. The bookmarks object contains stream identifiers as property names, each of which maps to an
+object describing the respective stream's state. As a concrete example:
+
+```
+{
+  "bookmarks": {
+    "orders": {
+      "last_record": "2017-07-07T10:20:00Z"
+    },
+    "customers": {
+      "last_record": 123
+    }
+  }
+}
+```
+
+In the above example, there are two streams that have been bookmarked: `"orders"` and `"customers"`.
+Each of those streams has some replication key field, like an `updated_at` timestamp or a sequential
+ID for append-only sources. The replication key value persisted as part of the state message allows
+the Tap to:
+
+- bookmark its progress through the stream
+- serve as a query parameter upon subsequent invocations, (e.g. `SELECT * from customers where id >
```

Comment:
I think this would help reduce confusion in the parenthetical: "e.g. the equivalent of, `SELECT * FROM customers WHERE id >= 123`"

(I also changed `>` to `>=`)